scripts/ai/utility/action/UtilityAIAction.gd

```diff
@@ -0,0 +1,35 @@
+class_name UtilityAIAction extends Node
+
+@export var action_name: StringName
+
+var considerations: Array[UtilityAIConsideration] = []
+
+var score: float = 0.0: set = set_score
+
+func _ready() -> void:
+	for child: Node in get_children():
+		if child is UtilityAIConsideration:
+			considerations.append(child)
+
+func set_score(s: float) -> void:
+	score = clampf(s, 0.0, 1.0)
```

Comment:
Do these scores eventually get compared per action to decide which one is taken?